src/contexts/auth-context.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { FC, useState, useEffect } from 'react';
+import {
+  User,
+  SignIn,
+  ServiceResolver,
+  ApiResponse,
+  JwtToken,
+  ErrorResponse,
+} from '@api';
+import { defaultProfileImage } from '@images';
+import { SessionStorageHelper, UserAuthHelper } from '@helpers';
+
+interface AuthContextState {
+  authenticated: boolean;
+  avatar: string;
+  member?: User;
+  signIn?: (
+    credentials: SignIn,
+  ) => Promise<ApiResponse<JwtToken | ErrorResponse>>;
+  signOut?: () => void;
+}
+
+const defaultAuthContextState: AuthContextState = {
+  authenticated: false,
+  avatar: defaultProfileImage,
+  member: undefined,
+};
+
+export const AuthContext = React.createContext(defaultAuthContextState);
+export const AuthConsumer = AuthContext.Consumer;
+
+export const AuthProvider: FC = (props) => {
+  const [authContextState, setAuthContextState] = useState<AuthContextState>(
```

Comment:
I think we let the TypeScript infer the type here. 

```suggestion
  const [authContextState, setAuthContextState] = useState(
```